### PR TITLE
Fix zero padding of bits in rainbow format flags (issue 549)

### PR DESCRIPTION
**Pull request by binomaiheu**
This is a fix for #549 . I've added a test for the truncation as well. Put in default argument for datashape = 0, it is only used when looking at the flags (datadepth < 8) and therefore code in the original tests for the radar data itself should remain unaffected.

### CODE REVIEW

File: wradlib/io/rainbow.py [hunk 1]
```diff
@@ -249,6 +251,9 @@ def map_rb_data(data, datadepth):
     if flagdepth:
         data = np.unpackbits(data)
 
+        # truncate excess bits
```
Comment (kmuehlbauer) [inline wradlib/io/rainbow.py:254]: Just do the subset on line 252.

File: wradlib/io/rainbow.py [hunk 2]
```diff
@@ -220,7 +220,7 @@ def get_rb_blob_data(datastring, blobid):
     return data
 
 
-def map_rb_data(data, datadepth):
+def map_rb_data(data, datadepth, datashape = 0):
```
Comment (kmuehlbauer) [inline wradlib/io/rainbow.py:223]: No spaces around `=`.

File: wradlib/tests/test_io.py [hunk 3]
```diff
@@ -1279,7 +1279,10 @@ def test_map_rb_data(self):
         np.testing.assert_allclose(io.rainbow.map_rb_data(indata, 32), outdata32)
         flagdata = b"1"
         np.testing.assert_allclose(
-            io.rainbow.map_rb_data(flagdata, 1), [0, 0, 1, 1, 0, 0, 0, 1]
+            io.rainbow.map_rb_data(flagdata, 1, 8), [0, 0, 1, 1, 0, 0, 0, 1]
+        )
+        np.testing.assert_allclose( # added test for truncation
```
Comment (kmuehlbauer) [inline wradlib/tests/test_io.py:1284]: move the comment directly above the call to np.testing...

File: wradlib/io/rainbow.py [hunk 4]
```diff
@@ -316,10 +321,12 @@ def get_rb_blob_from_string(datastring, blobdict):
     data = get_rb_blob_data(datastring, blobid)
     # map data to correct datatype and width
     datadepth = get_rb_data_attribute(blobdict, "depth")
-    data = map_rb_data(data, datadepth)
+
```
Comment (kmuehlbauer) [inline wradlib/io/rainbow.py:324]: remove empty line

**Review by kmuehlbauer — CHANGES_REQUESTED** [review]
@binomaiheu Thanks for the PR fixing this issue. I hope I've got all linter errors commented. Also have only one minor suggestion.

**Comment by kmuehlbauer** [comment]
@binomaiheu Good chance that `black` changed behaviour. Let me check first, what to do.

**Comment by kmuehlbauer** [comment]
@binomaiheu No worries, I'll take the rest from here.

**Comment by kmuehlbauer** [comment]
@binomaiheu I've squashed your changes into one commit. Your author information is kept within that commit, so your work is acknowledged correctly. 

It looks like there are some more changes needed for CI to work properly. Those are not related to your changes. I'll merge this when I've fixed things up.

**Comment by binomaiheu** [comment]
@kmuehlbauer, ok ! Thanks for pulling this together !

**Comment by kmuehlbauer** [comment]
Thanks @binomaiheu for the fix. And welcome as a first time contributor to wradlib. :tada:

**Comment by binomaiheu** [comment]
yay ;) hoping to be able to contribute more in future (still fairly new to the precip radar-field, but I'm learning)

**Comment by kmuehlbauer** [comment]
@binomaiheu Yes, that would be great! Looking forward to your next contributions. There are some issues open, which might be adequate.

**Pull request merged**